src/luagit2/libgit2/lua_libgit2.c

```diff
@@ -0,0 +1,259 @@
+#include "lua_libgit2.h"
+
+int lua_git_libgit2_init (lua_State *L) {
+	git_libgit2_init();
+	return 1;
+}
+
+int lua_git_libgit2_shutdown (lua_State *L) {
+	git_libgit2_shutdown();
+	return 1;
+}
+
+int lua_git_libgit2_features (lua_State *L) {
+	check_error_long(git_libgit2_features(),
+	    "Error Accessing libgit2 features", NULL);
+	int features_number = git_libgit2_features();
+	lua_pushinteger(L, features_number);
+	return 1;
+}
+
+int lua_git_libgit2_version (lua_State *L) {
+	libgit_version_data *version_data;
+
+	version_data = (libgit_version_data *)lua_newuserdata(L, sizeof(*version_data));
+	version_data->major  = 0;
+	version_data->minor = 0;
+	version_data->rev = 0;
+
+
+	luaL_getmetatable(L, "libgit2_version_data");
+
+	lua_setmetatable(L, -2);
+
+	int Major ;
+	int Minor ;
+	int Rev ;
+
+	git_libgit2_version(&Major, &Minor, &Rev );
+
+	version_data->major = Major;
+	version_data->minor = Minor;
+	version_data->rev = Rev;
+
+	return 1;
+
+}
+
+int lua_GIT_OPT_GET_MWINDOW_SIZE(lua_State *L) {
+	int *mmap_size ;
+	check_error_long(git_libgit2_opts(GIT_OPT_GET_MWINDOW_SIZE, mmap_size),
+	    "Error in getting memory map size", NULL);
+
+	int MMap_Size = *mmap_size;
```

Comment:
Is it necessary to do these two declarations? 

What if instead you had 
```
int MMap_Size ;
check_error_long(git_libgit2_opts(GIT_OPT_GET_MWINDOW_SIZE, &MMap_Size),
	    "Error in getting memory map size", NULL);
```
?

Reply:
I think i have this one fixed in later more updated branches :smile:

Reply:
ok! :)

Reply:
can you link it here? It will help me review :)

Reply:
https://github.com/SatyendraBanjare/luagit2/blob/008ccb286b5096b60904ec02712a03bafd714f77/src/luagit2/libgit2/lua_libgit2.c#L56